src/Microsoft.AspNet.Razor/Generator/TagHelperCodeGenerator.cs

```diff
@@ -75,9 +75,11 @@ public override void GenerateStartBlockCode(Block target, CodeGeneratorContext c
                 codeGenerator.Context.CodeTreeBuilder = new CodeTreeBuilder();
             }
 
+            var unprefixedTagName = tagHelperBlock.TagName.Substring(_tagHelperDescriptors.First().Prefix.Length);
```

Comment:
nit: though this is done only once per `TagHelperCodeGenerator`, might be slightly cleaner to set a `readonly int _prefixLength` field in the constructor